business/services.go

```diff
@@ -370,3 +370,47 @@ func (in *SvcService) getServiceValidations(services []core_v1.Service, deployme
 
 	return validations
 }
+
+// GetServiceAppName returns the "Application" name (app label) that relates to a service
+// This label is taken from pods, not the service object itself.
+// Consistency accross pods is assumed, ie. only one app label for all the pods under this service (only one will be returned).
+func (in *SvcService) GetServiceAppName(namespace, service string) (string, error) {
+	// Check if user has access to the namespace (RBAC) in cache scenarios and/or
+	// if namespace is accessible from Kiali (Deployment.AccessibleNamespaces)
+	if _, err := in.businessLayer.Namespace.GetNamespace(namespace); err != nil {
+		return "", err
+	}
+
+	svc, eps, err := in.getServiceDefinition(namespace, service)
+	if err != nil {
+		return "", err
+	}
+
+	labelsSelector := labels.Set(svc.Spec.Selector).String()
```

Comment:
Perhaps you can assume in Istio that service selector would contain the app level.
Because, if it's not also present in the Pods result would be the same (empty result).
So in that way no need to fetch pods and endpoints.
(I mean, the only corner case here would be that service selector don't use the app, but probably that would have other implications everywhere -> telemetry, istio config, etc).

Reply:
Well, I *think* we saw that case precisely of having services that don't select pods from the app label, but I'm not totally sure. And I don't think it causes any trouble in istio telemetry. I would prefer to play safe here, and since cache is generally enabled and very likely to already contains this data, the cost of this call should be minor.

Reply:
But look into the Service Details, it's using the selector for the very same scenario.

Reply:
I mean if we are using this pattern:
![image](https://user-images.githubusercontent.com/1662329/92230002-a5b25400-eeaa-11ea-93ae-dea7aa0462fc.png)

I guess using the service selector will just work and we can avoid that for one scenario people fetches the pod and for others the service and others the workload selector.

Also, if app is not present it will be marked with a validation as well.

Reply:
Ok, you're right, I didn't remember we were assuming this all throughout our code